src/run_program.rs

```diff
@@ -269,9 +297,11 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
 
     fn eval_pair(&mut self, program: NodePtr, env: NodePtr) -> Result<Cost, EvalErr> {
         #[cfg(feature = "pre-eval")]
-        if let Some(pre_eval) = &self.pre_eval {
-            if let Some(post_eval) = pre_eval(self.allocator, program, env)? {
-                self.posteval_stack.push(post_eval);
+        if let Some(pre_eval) = &mut self.pre_eval {
+            if let PreEvalResult::CallPostEval(pass) =
+                pre_eval.pre_eval(self.allocator, program, env)?
+            {
```

Comment:
`cargo fmt` prob needed

Reply:
PreEvalResult::CallPostEval is longer so it re-wraps a bit.  This is post fmt.

Reply:
Yeah fmt can be a bit funky at times, but that's the price of consistency I suppose :P

Reply:
Interesting, I don't recall having seen that lone `{` before.